spec/features/tool_drafts/forms/compatibility_and_usability_form_spec.rb

```diff
@@ -0,0 +1,162 @@
+describe 'Compatibility and Usability Form', js: true do
+  before do
+    login
+    draft = create(:empty_tool_draft, user: User.where(urs_uid: 'testuser').first)
+    visit edit_tool_draft_path(draft, 'compatibility_and_usability')
+
+    click_on 'Expand All'
+  end
+
+  context 'when viewing the form with no values' do
+    it 'displays the correct title' do
+      within 'header .collection-basics > h2' do
+        expect(page).to have_content('Compatibility and Usability')
+      end
+    end
+
+    it 'displays the form title in the breadcrumbs' do
+      within '.eui-breadcrumbs' do
+        expect(page).to have_content('Tool Drafts')
+        expect(page).to have_content('Compatibility and Usability')
+      end
+    end
+
+    it 'displays the correct sections' do
+      expect(page).to have_css('.eui-accordion__header > h3', text: 'Supported Formats')
+      expect(page).to have_css('.eui-accordion__header > h3', text: 'Supported Operating Systems')
+      expect(page).to have_css('.eui-accordion__header > h3', text: 'Supported Browsers')
+      expect(page).to have_css('.eui-accordion__header > h3', text: 'Supported Software Languages')
+      expect(page).to have_css('.eui-accordion__header > h3', text: 'Quality')
+      expect(page).to have_css('.eui-accordion__header > h3', text: 'Constraints')
+    end
+
+    it 'displays the correct number of required fields' do
```

Comment:
As I ran into some places where the required fields were not appearing correctly, do you think having a second test after entering data (and triggering required fields to recheck) per form is worth the cost?

Reply:
I will add a test in the individual form tests to also test for required icons again after filling out and saving.

Reply:
For the other part we discussed about this, we should think about how much to test or if we can figure out how to do it systematically via the schema as @jceaser suggested. Should we also have a ticket to determine this?

Reply:
I think that a ticket would be good. I think trying to drive it from the schema will be pretty difficult, but it would be good if we could.